Assets/Mirror/Runtime/Transport/Kcp/KcpTransport.cs

```diff
@@ -176,10 +179,10 @@ public override IEnumerable<Uri> ServerUri()
         public override async UniTask<IConnection> ConnectAsync(Uri uri)
         {
             var client = new KcpClientConnection();
-
             ushort port = (ushort)(uri.IsDefaultPort? Port : uri.Port);
 
             await client.ConnectAsync(uri.Host, port, HashCashBits);
+            client.SetNoDelayMode(delayMode);
```

Comment:
This is too late,  the client already sent the handshake.

This should be done inside client.ConnectAsync